core/fcache.c

```diff
@@ -3898,14 +3909,50 @@ chain_fragments_for_flush(dcontext_t *dcontext, fcache_unit_t *unit, fragment_t
     } else {
         ASSERT(list != NULL);
         ASSERT(prev_f != NULL);
-        prev_f->next_vmarea = NULL;
     }
 
     if (tail != NULL)
         *tail = prev_f;
     return list;
 }
 
+bool
+fcache_flush_fragment(dcontext_t *dcontext, app_pc tag)
+{
+    bool did_flush = false;
+    fragment_t *list_head = NULL;
+    DEBUG_DECLARE(bool flushed;)
+
+    /* we pass false to flush_fragments_in_region_start() below for owning the initexit
```

Comment:
style, here and below: capitalize; end punct; prefer `*/` on own line

Reply:
Done.